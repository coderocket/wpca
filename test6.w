x : int
  { x = X }
; if true -> x := -x [] true -> skip fi
  { x = abs[X] }

