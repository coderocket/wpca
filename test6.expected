open misc
open util/integer

pred true { no none }
pred false { some none }

one sig State {
 x : Int
}
one sig Const {
 X : Int
}
assert allguards_3_6 {
(State.x = Const.X => (true or (true or false)))
}
check allguards_3_6

assert guard_3_6 {
(State.x = Const.X => (true => (State.x.negate).(abs) = (Const.X).(abs)))
}
check guard_3_6

assert guard_4_6 {
(State.x = Const.X => (true => (State.x).(abs) = (Const.X).(abs)))
}
check guard_4_6

